test/unit/test_ext_auth.py

```diff
@@ -0,0 +1,264 @@
+        assert_that(storage.get_token_value("as")).is_equal_to((1, 2))
+        assert_that(storage.get_token_value("cappa")).is_equal_to(5)
+
+    @staticmethod
+    def test_token_remove():
+        storage = OneTimeTokenHandler(5)
+        storage.add_token(1, "cascata")
+        storage.get_token_value(1)
+        assert_that(storage.get_token_value(1)).is_none()
+
+
+def test_one_time_token_handler():
+    TestOneTimeTokenHandler()
+
+
+def test_token_generator():
+    assert_that(generate_random_token(256)).is_not_equal_to(generate_random_token(256))
+    assert_that(len(generate_random_token(128))).is_equal_to(128)
+
+
+def test_sha_generator():
+    assert_that(generate_sha512_hash("cavolo", "luca", 15)).is_equal_to(generate_sha512_hash("cavolo", "luca", 15))
+    assert_that(generate_sha512_hash("flower")).is_equal_to(generate_sha512_hash("flower"))
+    assert_that(generate_sha512_hash([1, 2, 4])).is_equal_to(generate_sha512_hash([1, 2, 4]))
+
+
+def _create_row(user, command, session_id):
+    return (
+        "{USER}                63   0.0  0.0  4348844   3108   ??  Ss   23Jul19   2:32.46 {COMMAND} "
+        "--session-id {SESSION}".format(USER=user, COMMAND=command, SESSION=session_id)
+    )
+
+
+def test_is_process_valid():
+    valid_command = "/usr/libexec/dcv/dcvagent"
+    invalid_command = "/usr/libexec/dcv/dcvagent2"
+    session_id = "mysession"
+    user = "luca"
+
+    assert_that(
+        DCVAuthenticator.is_process_valid(_create_row(user, valid_command, session_id), user, session_id)
+    ).is_true()
+    assert_that(
+        DCVAuthenticator.is_process_valid(_create_row(user, invalid_command, session_id), user, session_id)
+    ).is_false()
+    assert_that(
+        DCVAuthenticator.is_process_valid(_create_row("abcd", valid_command, session_id), user, session_id)
+    ).is_false()
+    assert_that(
+        DCVAuthenticator.is_process_valid(_create_row(user, valid_command, "afsd"), user, session_id)
+    ).is_false()
+
+
+def mock_generate_random_token(mocker, value):
+    mocker.patch(MAIN + "generate_random_token", return_value=value)
+
+
+def mock_verify_session_existence(mocker, exists):
+    def _return_value(user, sessionid):
+        if not exists:
+            raise DCVAuthenticator.IncorrectRequestException("The given session for the user does not exists")
+
+    mocker.patch(EXT_AUTH + "_verify_session_existence", side_effect=_return_value)
+
+
+def mock_os(mocker, user, timestamp):
+    class FileProperty(object):
+        pass
+
+    file_property = FileProperty
+    file_property.st_uid = 1
+    file_property.st_size = b"800"
+    file_property.st_mtime = timestamp
+    file_property.pw_name = user
+    file_property.st_mode = 0
+
+    mocker.patch(MAIN + "os.stat", return_value=file_property, autospec=True)
+    mocker.patch(MAIN + "os.remove")
+    mocker.patch(MAIN + "getpwuid", autospec=True, return_value=file_property)
+
+
+def test_get_request_token_parameter():
```

Comment:
you could use [@pytest.mark.parametrize](https://docs.pytest.org/en/latest/parametrize.html)
it is very helpful for a set of tests like the following.